{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "\n",
    "# Function to plot the coverage per base of a genome with and without zoom\n",
    "def coverage_graph(input, output, amp_start, amp_end, zoom_start=None, zoom_end=None, title=None, total_coverage=None):\n",
    "    df = pd.read_csv(input, sep='\\t', names=['genome','base', 'coverage'])\n",
    "\n",
    "    sns.lineplot(data=df, x='base', y='coverage')\n",
    "    plt.title(title + '\\n' + 'Total coverage ' + total_coverage)\n",
    "    plt.xlabel('Nucleotide position')\n",
    "    plt.ylabel('Coverage')\n",
    "    plt.axvline(x=amp_start, color='red', linestyle='--')\n",
    "    plt.axvline(x=amp_end, color='red', linestyle='--')\n",
    "    plt.savefig(output)\n",
    "    plt.savefig(output.replace('.png', '.svg'))\n",
    "    plt.clf()\n",
    "\n",
    "    if zoom_start and zoom_end:\n",
    "        sns.lineplot(data=df, x='base', y='coverage')\n",
    "        plt.xlim(zoom_start, zoom_end)\n",
    "        plt.title(title + '\\n' + 'Total coverage: ' + total_coverage)\n",
    "        plt.xlabel('Nucleotide position')\n",
    "        plt.ylabel('Coverage')\n",
    "        plt.axvline(x=amp_start, color='red', linestyle='--')\n",
    "        plt.axvline(x=amp_end, color='red', linestyle='--')\n",
    "        plt.savefig(output.replace('.png', '_zoom.png'))\n",
    "        plt.savefig(output.replace('.png', '_zoom.svg'))\n",
    "        plt.clf()\n",
    "        \n",
    "# Execution of the function for each genome\n",
    "# coverage_graph(input='./01_besthit_Caudorviricetes_sp._isolate_ctJ1L4_viral_allsamples_aligned_genomecov.txt',\n",
    "#                output='./01_besthit_Caudorviricetes_sp._isolate_ctJ1L4_viral_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=19543,\n",
    "#                amp_end=19898,\n",
    "#             #    zoom_start=9526,\n",
    "#             #    zoom_end=29989,\n",
    "#                title='Viral coverage: Caudorviricetes sp. isolate ctJ1L4',\n",
    "#                total_coverage='70.46%')\n",
    "\n",
    "# coverage_graph(input='./02_besthit_Caudoviricetes_sp._isolate_ctzDR1_viral_allsamples_aligned_genomecov.txt',\n",
    "#                output='./02_besthit_Caudoviricetes_sp._isolate_ctzDR1_viral_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=15016,\n",
    "#                amp_end=15374,\n",
    "#             #    zoom_start=4999,\n",
    "#             #    zoom_end=25462,\n",
    "#                title='Caudoviricetes sp. isolate ctzDR1',\n",
    "#                total_coverage='with Viral DNA reads: 84.86%')\n",
    "\n",
    "# coverage_graph(input='./02_besthit_Caudoviricetes_sp._isolate_ctzDR1_allsamples_aligned_genomecov.txt',\n",
    "#                output='./02_besthit_Caudoviricetes_sp._isolate_ctzDR1_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=15016,\n",
    "#                amp_end=15374,\n",
    "#             #    zoom_start=4999,\n",
    "#             #    zoom_end=25462,\n",
    "#                title='Caudoviricetes sp. isolate ctzDR1',\n",
    "#                total_coverage='with RNA reads: 13.39%')\n",
    "\n",
    "# coverage_graph(input='./03_besthit_Anaerotignum_sp._MB30-C6_viral_allsamples_aligned_genomecov.txt',\n",
    "#                output='./03_besthit_Anaerotignum_sp._MB30-C6_viral_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=2477315,\n",
    "#                amp_end=2477669,\n",
    "#             #    zoom_start=2467298,\n",
    "#             #    zoom_end=2487761,\n",
    "#                title='Viral coverage: Anaerotignum sp. MB30-C6',\n",
    "#                total_coverage='1.85%')\n",
    "\n",
    "# coverage_graph(input='./04_besthit_Anaerotignum_propionicum_DSM_1682_viral_allsamples_aligned_genomecov.txt',\n",
    "#                output='./04_besthit_Anaerotignum_propionicum_DSM_1682_viral_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=2496741,\n",
    "#                amp_end=2497093,\n",
    "#             #    zoom_start=2486724,\n",
    "#             #    zoom_end=2507184,\n",
    "#                title='Viral coverage: Anaerotignum propionicum DSM 1682',\n",
    "#                total_coverage='1.83%')\n",
    "\n",
    "# coverage_graph(input='./05_besthit_Clostridium_sp._MD294_strain_ASF356_viral_allsamples_aligned_genomecov.txt',\n",
    "#                output='./05_besthit_Clostridium_sp._MD294_strain_ASF356_viral_allsamples_aligned_genomecov.png',\n",
    "#                amp_start=1388458,\n",
    "#                amp_end=1388805,\n",
    "#             #    zoom_start=1378441,\n",
    "#             #    zoom_end=1398904,\n",
    "#                title='Viral coverage: Clostridium sp. MD294 strain ASF356',\n",
    "#                total_coverage='1.77%')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:24: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='reads', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:84: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='relative_abundance', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:144: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='log10', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:204: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='coverage', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:24: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='reads', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:84: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='relative_abundance', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:144: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='log10', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:204: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='coverage', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:24: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='reads', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:84: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='relative_abundance', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:144: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='log10', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:204: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='coverage', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:24: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='reads', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:84: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='relative_abundance', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:144: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='log10', showfliers=False, palette=colors)\n",
      "/var/folders/v4/vr3wk_q55dbbq68v6wdr2mrh0000gn/T/ipykernel_46192/3633987475.py:204: FutureWarning: \n",
      "\n",
      "Passing `palette` without assigning `hue` is deprecated and will be removed in v0.14.0. Assign the `x` variable to `hue` and set `legend=False` for the same effect.\n",
      "\n",
      "  ax = sns.boxplot(data=df, x='group', y='coverage', showfliers=False, palette=colors)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import os\n",
    "# %pip install scipy\n",
    "from scipy.stats import mannwhitneyu\n",
    "\n",
    "# Boxplot function to plot number of reads per sample\n",
    "def boxplot(input, title):\n",
    "    df = pd.read_csv(input, sep='\\t')\n",
    "\n",
    "    # Add group column to the dataframe\n",
    "    df['group'] = df['sample'].apply(lambda x: 'H' if 'H' in x else ('OMC' if 'OMC' in x else 'O'))\n",
    "\n",
    "    # Define colors for the b# Remove the H group\n",
    "    df = df[df['group'] != 'H']\n",
    "\n",
    "    # Define colors for the boxplot\n",
    "    # colors = ['blue', 'orange', 'red']\n",
    "    colors = ['orange', 'red']\n",
    "\n",
    "    # Boxplot with reads aligned per sample\n",
    "    ax = sns.boxplot(data=df, x='group', y='reads', showfliers=False, palette=colors)\n",
    "    sns.stripplot(data=df, x='group', y='reads', color='black', size=4)\n",
    "    plt.title(title)\n",
    "    plt.xlabel('')\n",
    "    plt.ylabel('Reads aligned')\n",
    "\n",
    "    # Perform mannwhitneyu analysis\n",
    "    # group_H = df[df['group'] == 'H']['reads']\n",
    "    group_OMC = df[df['group'] == 'OMC']['reads']\n",
    "    group_O = df[df['group'] == 'O']['reads']\n",
    "\n",
    "    # Compare group H and OMC\n",
    "    # u, p_H_OMC = mannwhitneyu(group_H, group_OMC)\n",
    "\n",
    "    # Compare group H and O\n",
    "    # u, p_H_O = mannwhitneyu(group_H, group_O)\n",
    "\n",
    "    # Compare group OMC and O\n",
    "    u, p_OMC_O = mannwhitneyu(group_OMC, group_O)\n",
    "\n",
    "    # Get the y-axis limits\n",
    "    bottom, top = ax.get_ylim()\n",
    "    y_range = top - bottom\n",
    "\n",
    "    # Significance bars\n",
    "    # significant_combinations = [((0, 2), p_H_OMC), ((0, 1), p_H_O), ((1, 2), p_OMC_O)]\n",
    "    significant_combinations = [((0, 1), p_OMC_O)]\n",
    "    for i, significant_combination in enumerate(significant_combinations):\n",
    "        # Columns corresponding to the datasets of interest\n",
    "        x1 = significant_combination[0][0]\n",
    "        x2 = significant_combination[0][1]\n",
    "        # What level is this bar among the bars above the plot?\n",
    "        level = len(significant_combinations) - i\n",
    "        # Plot the bar\n",
    "        bar_height = (y_range * 0.07 * level) + top\n",
    "        bar_tips = bar_height - (y_range * 0.02)\n",
    "        plt.plot(\n",
    "            [x1, x1, x2, x2],\n",
    "            [bar_tips, bar_height, bar_height, bar_tips], lw=1, c='k'\n",
    "        )\n",
    "        # Significance level\n",
    "        p = significant_combination[1]\n",
    "        if p < 0.001:\n",
    "            sig_symbol = '***'\n",
    "        elif p < 0.01:\n",
    "            sig_symbol = '**'\n",
    "        elif p < 0.05:\n",
    "            sig_symbol = '*'\n",
    "        else:\n",
    "            sig_symbol = 'n.s.'\n",
    "        text_height = bar_height + (y_range * 0.01)\n",
    "        plt.text((x1 + x2) * 0.5, text_height, sig_symbol, ha='center', va='bottom', c='k')\n",
    "\n",
    "    output = os.path.splitext(input)[0] + '.png'\n",
    "    plt.savefig(output)\n",
    "    output = os.path.splitext(input)[0] + '.svg'\n",
    "    plt.savefig(output)\n",
    "    plt.clf()\n",
    "\n",
    "    # Boxplot with relative abundance per sample\n",
    "    ax = sns.boxplot(data=df, x='group', y='relative_abundance', showfliers=False, palette=colors)\n",
    "    sns.stripplot(data=df, x='group', y='relative_abundance', color='black', size=4)\n",
    "    plt.title(title)\n",
    "    plt.xlabel('')\n",
    "    plt.ylabel('Relative abundance (%)')\n",
    "\n",
    "    # Perform mannwhitneyu analysis\n",
    "    # group_H = df[df['group'] == 'H']['relative_abundance']\n",
    "    group_OMC = df[df['group'] == 'OMC']['relative_abundance']\n",
    "    group_O = df[df['group'] == 'O']['relative_abundance']\n",
    "\n",
    "    # Compare group H and OMC\n",
    "    # u, p_H_OMC = mannwhitneyu(group_H, group_OMC)\n",
    "\n",
    "    # Compare group H and O\n",
    "    # u, p_H_O = mannwhitneyu(group_H, group_O)\n",
    "\n",
    "    # Compare group OMC and O\n",
    "    u, p_OMC_O = mannwhitneyu(group_OMC, group_O)\n",
    "\n",
    "    # Get the y-axis limits\n",
    "    bottom, top = ax.get_ylim()\n",
    "    y_range = top - bottom\n",
    "\n",
    "    # Significance bars\n",
    "    # significant_combinations = [((0, 2), p_H_OMC), ((0, 1), p_H_O), ((1, 2), p_OMC_O)]\n",
    "    significant_combinations = [((0, 1), p_OMC_O)]\n",
    "    for i, significant_combination in enumerate(significant_combinations):\n",
    "        # Columns corresponding to the datasets of interest\n",
    "        x1 = significant_combination[0][0]\n",
    "        x2 = significant_combination[0][1]\n",
    "        # What level is this bar among the bars above the plot?\n",
    "        level = len(significant_combinations) - i\n",
    "        # Plot the bar\n",
    "        bar_height = (y_range * 0.07 * level) + top\n",
    "        bar_tips = bar_height - (y_range * 0.02)\n",
    "        plt.plot(\n",
    "            [x1, x1, x2, x2],\n",
    "            [bar_tips, bar_height, bar_height, bar_tips], lw=1, c='k'\n",
    "        )\n",
    "        # Significance level\n",
    "        p = significant_combination[1]\n",
    "        if p < 0.001:\n",
    "            sig_symbol = '***'\n",
    "        elif p < 0.01:\n",
    "            sig_symbol = '**'\n",
    "        elif p < 0.05:\n",
    "            sig_symbol = '*'\n",
    "        else:\n",
    "            sig_symbol = 'n.s.'\n",
    "        text_height = bar_height + (y_range * 0.01)\n",
    "        plt.text((x1 + x2) * 0.5, text_height, sig_symbol, ha='center', va='bottom', c='k')\n",
    "\n",
    "    output = os.path.splitext(input)[0] + '_relative_abundance.png'\n",
    "    plt.savefig(output)\n",
    "    output = os.path.splitext(input)[0] + '_relative_abundance.svg'\n",
    "    plt.savefig(output)\n",
    "    plt.clf()\n",
    "\n",
    "    # Boxplot with log10 abundance per sample\n",
    "    ax = sns.boxplot(data=df, x='group', y='log10', showfliers=False, palette=colors)\n",
    "    sns.stripplot(data=df, x='group', y='log10', color='black', size=4)\n",
    "    plt.title(title)\n",
    "    plt.xlabel('')\n",
    "    plt.ylabel('log10 abundance')\n",
    "\n",
    "    # Perform mannwhitneyu analysis\n",
    "    # group_H = df[df['group'] == 'H']['log10']\n",
    "    group_OMC = df[df['group'] == 'OMC']['log10']\n",
    "    group_O = df[df['group'] == 'O']['log10']\n",
    "\n",
    "    # Compare group H and OMC\n",
    "    # u, p_H_OMC = mannwhitneyu(group_H, group_OMC)\n",
    "\n",
    "    # Compare group H and O\n",
    "    # u, p_H_O = mannwhitneyu(group_H, group_O)\n",
    "\n",
    "    # Compare group OMC and O\n",
    "    u, p_OMC_O = mannwhitneyu(group_OMC, group_O)\n",
    "\n",
    "    # Get the y-axis limits\n",
    "    bottom, top = ax.get_ylim()\n",
    "    y_range = top - bottom\n",
    "\n",
    "    # Significance bars\n",
    "    # significant_combinations = [((0, 2), p_H_OMC), ((0, 1), p_H_O), ((1, 2), p_OMC_O)]\n",
    "    significant_combinations = [((0, 1), p_OMC_O)]\n",
    "    for i, significant_combination in enumerate(significant_combinations):\n",
    "        # Columns corresponding to the datasets of interest\n",
    "        x1 = significant_combination[0][0]\n",
    "        x2 = significant_combination[0][1]\n",
    "        # What level is this bar among the bars above the plot?\n",
    "        level = len(significant_combinations) - i\n",
    "        # Plot the bar\n",
    "        bar_height = (y_range * 0.07 * level) + top\n",
    "        bar_tips = bar_height - (y_range * 0.02)\n",
    "        plt.plot(\n",
    "            [x1, x1, x2, x2],\n",
    "            [bar_tips, bar_height, bar_height, bar_tips], lw=1, c='k'\n",
    "        )\n",
    "        # Significance level\n",
    "        p = significant_combination[1]\n",
    "        if p < 0.001:\n",
    "            sig_symbol = '***'\n",
    "        elif p < 0.01:\n",
    "            sig_symbol = '**'\n",
    "        elif p < 0.05:\n",
    "            sig_symbol = '*'\n",
    "        else:\n",
    "            sig_symbol = 'n.s.'\n",
    "        text_height = bar_height + (y_range * 0.01)\n",
    "        plt.text((x1 + x2) * 0.5, text_height, sig_symbol, ha='center', va='bottom', c='k')\n",
    "\n",
    "    output = os.path.splitext(input)[0] + '_log10.png'\n",
    "    plt.savefig(output)\n",
    "    output = os.path.splitext(input)[0] + '_log10.svg'\n",
    "    plt.savefig(output)\n",
    "    plt.clf()\n",
    "\n",
    "    # Boxplot with log10 abundance per sample\n",
    "    ax = sns.boxplot(data=df, x='group', y='coverage', showfliers=False, palette=colors)\n",
    "    sns.stripplot(data=df, x='group', y='coverage', color='black', size=4)\n",
    "    plt.title(title)\n",
    "    plt.xlabel('')\n",
    "    plt.ylabel('Total coverage (%)')\n",
    "\n",
    "    # Perform mannwhitneyu analysis\n",
    "    # group_H = df[df['group'] == 'H']['coverage']\n",
    "    group_OMC = df[df['group'] == 'OMC']['coverage']\n",
    "    group_O = df[df['group'] == 'O']['coverage']\n",
    "\n",
    "    # Compare group H and OMC\n",
    "    # u, p_H_OMC = mannwhitneyu(group_H, group_OMC)\n",
    "\n",
    "    # Compare group H and O\n",
    "    # u, p_H_O = mannwhitneyu(group_H, group_O)\n",
    "\n",
    "    # Compare group OMC and O\n",
    "    u, p_OMC_O = mannwhitneyu(group_OMC, group_O)\n",
    "\n",
    "    # Get the y-axis limits\n",
    "    bottom, top = ax.get_ylim()\n",
    "    y_range = top - bottom\n",
    "\n",
    "    # Significance bars\n",
    "    # significant_combinations = [((0, 2), p_H_OMC), ((0, 1), p_H_O), ((1, 2), p_OMC_O)]\n",
    "    significant_combinations = [((0, 1), p_OMC_O)]\n",
    "    for i, significant_combination in enumerate(significant_combinations):\n",
    "        # Columns corresponding to the datasets of interest\n",
    "        x1 = significant_combination[0][0]\n",
    "        x2 = significant_combination[0][1]\n",
    "        # What level is this bar among the bars above the plot?\n",
    "        level = len(significant_combinations) - i\n",
    "        # Plot the bar\n",
    "        bar_height = (y_range * 0.07 * level) + top\n",
    "        bar_tips = bar_height - (y_range * 0.02)\n",
    "        plt.plot(\n",
    "            [x1, x1, x2, x2],\n",
    "            [bar_tips, bar_height, bar_height, bar_tips], lw=1, c='k'\n",
    "        )\n",
    "        # Significance level\n",
    "        p = significant_combination[1]\n",
    "        if p < 0.001:\n",
    "            sig_symbol = '***'\n",
    "        elif p < 0.01:\n",
    "            sig_symbol = '**'\n",
    "        elif p < 0.05:\n",
    "            sig_symbol = '*'\n",
    "        else:\n",
    "            sig_symbol = 'n.s.'\n",
    "        text_height = bar_height + (y_range * 0.01)\n",
    "        plt.text((x1 + x2) * 0.5, text_height, sig_symbol, ha='center', va='bottom', c='k')\n",
    "\n",
    "    output = os.path.splitext(input)[0] + '_coverage.png'\n",
    "    plt.savefig(output)\n",
    "    output = os.path.splitext(input)[0] + '_coverage.svg'\n",
    "    plt.savefig(output)\n",
    "    plt.clf()\n",
    "\n",
    "# # Execution of the function with RNA reads\n",
    "boxplot(input='./reads_aligned_per_sample_Caudorviricetes_sp._isolate_ctJ1L4_rna.tsv',\n",
    "        title='Caudoviricetes sp. isolate ctJ1L4\\nRNA reads aligned per sample ')\n",
    "\n",
    "# # Execution of the function with RNA reads\n",
    "boxplot(input='./reads_aligned_per_sample_Caudoviricetes_sp._isolate_ctzDR1_rna.tsv',\n",
    "        title='Caudoviricetes sp. isolate ctzDR1\\nRNA reads aligned per sample')\n",
    "\n",
    "\n",
    "# Execution of the function with DNA VLPs reads\n",
    "boxplot(input='./reads_aligned_per_sample_Caudorviricetes_sp._isolate_ctJ1L4_dna.tsv',\n",
    "        title='Caudoviricetes sp. isolate ctJ1L4\\nDNA reads aligned per sample')\n",
    "\n",
    "boxplot(input='./reads_aligned_per_sample_Caudoviricetes_sp._isolate_ctzDR1_dna.tsv',\n",
    "        title='Caudoviricetes sp. isolate ctzDR1\\nDNA reads aligned per sample')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
